src/otx/algo/common/utils/bbox_overlaps.py

```diff
@@ -142,15 +144,28 @@ def bbox_overlaps(
         >>> assert tuple(bbox_overlaps(nonempty, empty).shape) == (1, 0)
         >>> assert tuple(bbox_overlaps(empty, empty).shape) == (0, 0)
     """
+    if not (bboxes1.size(-1) == 4 or bboxes1.size(0) == 0):
+        msg = "bboxes1 must have a last dimension of size 4 or be an empty tensor."
+        raise ValueError(msg)
+
+    if not (bboxes2.size(-1) == 4 or bboxes2.size(0) == 0):
+        msg = "bboxes2 must have a last dimension of size 4 or be an empty tensor."
+        raise ValueError(msg)
+
+    if bboxes1.shape[:-2] != bboxes2.shape[:-2]:
+        msg = "The batch dimension of bboxes must be the same."
+        raise ValueError(msg)
+
     batch_shape = bboxes1.shape[:-2]
 
     rows = bboxes1.size(-2)
     cols = bboxes2.size(-2)
 
     if rows * cols == 0:
+        warnings.warn("No bboxes are provided! Returning empty boxes!", stacklevel=2)
         if is_aligned:
-            return bboxes1.new((*batch_shape, rows))
-        return bboxes1.new((*batch_shape, rows, cols))
+            return bboxes1.new(batch_shape + (rows,))  # noqa: RUF005
+        return bboxes1.new(batch_shape + (rows, cols))  # noqa: RUF005
```

Comment:
FYI. It can be conflicted with https://github.com/openvinotoolkit/training_extensions/pull/4003, could you discuss it with @harimkang ?

Reply:
After this PR is merged, I'll take another look at it from my PR end.